### DISCUSSION TITLE
Ignition multisig integration

### DISCUSSION THREAD

**Discussion by marcelomorgado** [NomicFoundation/hardhat#5533] [Troubleshooting]
**TL;DR**
Hi, I would like to know how would be the best way to integrate `ignition` with a `multisig` wallet (i.e., `safe`).

**Details**
For instance, the contracts' owner (`proxyAdmin` included) is a multisig wallet. What I'm doing right now with `hardhat-deploy` is the following:

1. Scripts run
2. When some call is a multisig (i.e. `{ from: MULSIG_ADDR }`) two things can happen:
  2.1. The following script execution depends on this specific call (e.g., the call is a `upgradeTo` call and after that there is a call to a new implemented function)
  2.2. The multisig call isn't a dependency for further calls
  If `2.1.` the multisig call is proposed and the script is aborted waiting for its execution
  Otherwise if `2.2.` the multisig call is stored in a file and the execution continues
3. The script continues (manually after abortion or keep executing)
4. At the end, if there is any stored multisig call, they are batch and proposed

I'm not sure if it's the best possible approach but it has being work well for many complex scenarios I've faced so far.
That all said, my question is what would be the best way to having such integration from `ignition`?

**Comment by kanej** [comment]
Ignition does not have built in support for multisig wallets, though it is on our roadmap.
Our intention is to support a `safe` strategy, that will proxy deployments and calls through the multisig, halting the execution awaiting sign-off.

For the moment, you will end up falling back to a Hardhat script calling out to Ignition for sub-parts of the deployment.
